dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumApplicationScopeTest.kt

```diff
@@ -121,16 +150,196 @@ internal class RumApplicationScopeTest {
     fun `M return true W isActive()`() {
         val isActive = testedScope.isActive()
 
-        assertThat(isActive).isTrue()
+        assertThat(isActive).isTrue
     }
 
     @Test
     fun `delegates all events to child scope`() {
-        testedScope.setFieldValue("childScope", mockChildScope)
+        testedScope.childScopes.clear()
+        testedScope.childScopes.add(mockChildScope)
 
         testedScope.handleEvent(mockEvent, mockWriter)
 
         verify(mockChildScope).handleEvent(mockEvent, mockWriter)
         verifyZeroInteractions(mockWriter)
     }
+
+    @Test
+    fun `M have return active session W activeSession`() {
+        // Then
+        val activeSession = testedScope.activeSession
+        check(activeSession is RumSessionScope)
+    }
+
+    @Test
+    fun `M have no active session W stopping current session`() {
+        // When
+        testedScope.handleEvent(RumRawEvent.StopSession(), mockWriter)
+
+        // Then
+        val activeSession = testedScope.activeSession
+        assertThat(activeSession).isNull()
+    }
+
+    @Test
+    fun `M keep inactiveSession until completed W handleEvent`() {
+        // Given
+        val mockSession: RumSessionScope = mock()
+        testedScope.childScopes.clear()
+        testedScope.childScopes.add(mockSession)
+        val stopEvent = RumRawEvent.StopSession()
+        whenever(mockSession.handleEvent(stopEvent, mockWriter)) doReturn mockSession
+
+        // When
+        testedScope.handleEvent(stopEvent, mockWriter)
+
+        // Then
+        assertThat(testedScope.childScopes).isNotEmpty
+        assertThat(testedScope.childScopes.first()).isEqualTo(mockSession)
+    }
+
+    @Test
+    fun `M create a new session W handleEvent { no active sessions, user interaction } `(
+        @StringForgery viewKey: String,
+        @StringForgery viewName: String
+    ) {
+        // Given
+        val initialSession = testedScope.childScopes.first()
+        testedScope.handleEvent(RumRawEvent.StopSession(), mockWriter)
+        testedScope.handleEvent(
+            RumRawEvent.StartView(
+                key = viewKey,
+                name = viewName,
+                attributes = mapOf()
+            ),
+            mockWriter
+        )
+
+        // Then
+        assertThat(testedScope.childScopes.count()).isEqualTo(1)
+        val newSession = testedScope.childScopes.first()
+        check(newSession is RumSessionScope)
+        assertThat(newSession).isNotSameAs(initialSession)
+        assertThat(newSession.samplingRate).isEqualTo(fakeSamplingRate)
+        assertThat(newSession.backgroundTrackingEnabled).isEqualTo(fakeBackgroundTrackingEnabled)
+        assertThat(newSession.firstPartyHostHeaderTypeResolver).isSameAs(mockResolver)
+    }
+
+    @Test
+    fun `M create a new session with last known view W handleEvent { no active sessions, start action } `(
```

Comment:
I'm not quite clear why do we create a new session with last known view. Can you please elaborate?

Reply:
I think that was just what we decided to do. The requirements require that we start a new session in response to a user interaction, but in order to get any information from a user action, they have to be attached to a view. So, if the session is starting from an action, we take the last known view and restart it.